### DISCUSSION TITLE
No package metadata was found for email-validator

### DISCUSSION THREAD

**Discussion by dwbelliston** [pydantic/pydantic#9588] [Question]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

Hey! Thanks for the work on this project. Been running well on version 1, just tried to upgrade to v2 and having some issues.

I am deploying to aws lambda and a very simple works.

Then when i try to use EmailStr and I get an issue saying i need to `pip install pydantic[email]`

Once i do that... i get an error:

```
[ERROR] Runtime.ImportModuleError: Unable to import module 'app.main': No package metadata was found for email-validator
```

I checked the code and the email-validator dep is in the lambda code. I have tried quite a few different approaches, but keep getting this metadata issue.

Would love suggestions on other ideas to debug this. Thanks!!

### Example Code

```Python
"""
API

Main entrypoint to API
"""

from pydantic import BaseModel, EmailStr

from fastapi import FastAPI
from mangum import Mangum


# Properties to receive via API on creation
class SampleModel(BaseModel):
    email: EmailStr


# Set up app
app = FastAPI(
    title="SAMPLE_NAME",
)


@app.get("/test", response_model=SampleModel)
async def trigger_error():

    return SampleModel(email="test@me.com")
    # division_by_zero = 1 / 0


# Wrap in ASGI to deploy on lambda
handler = Mangum(app)  # type: ignore
```

### Python, Pydantic & OS Version

```Text
running in aws lambda

runtime python 3.11
```

**Comment by sydney-runkle** [comment]
@dwbelliston,

Have you tried `pip install email-validator`? That should resolve the issue!

Marking as resolved for now, but feel free to ping me if you're still having trouble!

**Comment by dwbelliston** [comment]
Thanks for the help

Yes, I have put  `email-validator` in my requirements.txt for my deploy to AWS lambda. When I download the zip, I do see the email-validator package alongside pydantic.

`No package metadata` thats the issue I am getting. As opposed to module not found. So I do believe its being installed, something is off though and I have not been able to figure it out.

**Comment by sydney-runkle** [comment]
@dwbelliston,

Huh, that's quite odd. What happens when you run said code in a local environment?

**Reply by dwbelliston** [reply]
Ya so it does work locally, but its packaged a little bit different to get things into a docker container, whereas when I delploy to lambda, thats just built and zipped on github actions. So.. could be several things in there that could be different.

Im just going to drop the use of EmailStr for now and move on and try again on later verisons.

I did wonder if it has to do with this, but not amazing at how python deps work so may be irrelevant:
https://github.com/JoshData/python-email-validator/issues/138

**Comment by joejacobs** [comment]
Having the exact same issue here. Works fine when I use pydantic locally on my mac. However, when I attempt to deploy the same code to a lambda, I get the above error.

**Comment by lukasz-kilaszewski** [comment]
In my case the same problem. Locally everything works great, when the code is deployed to lambda the same error appears.

**Comment by ngorgol** [comment]
Same problem when I try to deploy my to lambda. I'm blocked now to deploy my FastAPI on lambda for my project...
![image](https://github.com/pydantic/pydantic/assets/142744331/f5bda43d-991b-42cf-8222-5e1f299beec5)

**Comment by ngorgol** [comment]
Is there at least a workaround for short-term ?

**Comment by sydney-runkle** [comment]
Here's a solution: https://github.com/pydantic/pydantic/pull/6033#issuecomment-1927020910.

Generally, an issue with aws lambda and package metadata.

**Reply by matthoendorf** [reply]
In addition to setting `slim` to `false`, I had to run `serverless requirements cleanCache` in order for this solution to work.

See https://github.com/serverless/serverless-python-requirements/issues/437#issuecomment-741629169

**Comment by dwbelliston** [comment]
thank you!!

**Comment by sydney-runkle** [comment]
I've opened a PR with a note in the docs here, should be merged soon!

https://github.com/pydantic/pydantic/pull/9725

**Comment by sudarshan-uprety** [comment]
is this solved for aws lambda? my serverless API is not working saying "No package metadata was found for email-validator"

**Reply by sydney-runkle** [reply]
See https://docs.pydantic.dev/latest/integrations/aws_lambda/
